### ISSUE TITLE
SSLForceHost does not work as expected

### ISSUE THREAD

**Issue by m-nouredini** [containous/traefik#4087]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

I run a container with the following labels:
```
version: '3'

services:
  web:
    image: website:latest
    networks:
      - web
    labels:
      - "traefik.docker.network=web"
      - "traefik.enable=true"
      - "traefik.frontend.rule=Host:foo.example.com,www.foo.example.com"
      - "traefik.port=80"
      - "traefik.protocol=http"
      - "traefik.frontend.headers.SSLRedirect=true"
      - "traefik.frontend.headers.SSLForceHost=true"
      - "traefik.frontend.headers.SSLHost=foo.example.com"

networks:
  web:
    external: true

```

### What did you expect to see?

When I curl https://www.foo.example.com I get 301 redirect to https://foo.example.com

### What did you see instead?

I get 200 response and the request is not redirected

### Output of `traefik version`:

```
Version:      v1.7.3
Codename:     maroilles
Go version:   go1.11.1
Built:        2018-10-15_10:13:00AM
OS/Arch:      linux/amd64
```

### What is your environment & configuration?

```toml
debug = true

defaultEntryPoints = ["https","http"]

[api]
  dashboard = true

[entryPoints]
  [entryPoints.http]
  address = ":80"
  [entryPoints.https]
  address = ":443"
  [entryPoints.https.tls]

[retry]

[docker]
endpoint = "unix:///var/run/docker.sock"
domain = "example.com"
watch = true
exposedByDefault = false

[acme]
email = "dev@example.com"
storage = "acme.json"
onHostRule = true
caServer = "https://acme-v02.api.letsencrypt.org/directory"
entryPoint = "https"
  [acme.httpChallenge]
  entryPoint = "http"

[[acme.domains]]
  main = "foo.example.com"

```

### The log output in DEBUG level:

```

time="2018-10-21T15:30:13Z" level=info msg="Using TOML configuration file /traefik.toml"
time="2018-10-21T15:30:13Z" level=info msg="Traefik version v1.7.3 built on 2018-10-15_10:13:00AM"
time="2018-10-21T15:30:13Z" level=debug msg="Global configuration loaded {"LifeCycle":{"RequestAcceptGraceTimeout":0,"GraceTimeOut":10000000000},"GraceTimeOut":0,"Debug":true,"CheckNewVersion":true,"SendAnonymousUsage":false,"AccessLogsFile":"","AccessLog":null,"TraefikLogsFile":"","TraefikLog":null,"Tracing":null,"LogLevel":"","EntryPoints":{"http":{"Address":":80","TLS":null,"Redcomect":null,"Auth":null,"WhitelistSourceRange":null,"WhiteList":null,"Compress":false,"ProxyProtocol":null,"ForwardedHeaders":{"Insecure":true,"TrustedIPs":null}},"https":{"Address":":443","TLS":{"MinVersion":"","CipherSuites":null,"Certificates":null,"ClientCAFiles":null,"ClientCA":{"Files":null,"Optional":false},"DefaultCertificate":null,"SniStrict":false},"Redcomect":null,"Auth":null,"WhitelistSourceRange":null,"WhiteList":null,"Compress":false,"ProxyProtocol":null,"ForwardedHeaders":{"Insecure":true,"TrustedIPs":null}},"traefik":{"Address":":8080","TLS":null,"Redcomect":null,"Auth":null,"WhitelistSourceRange":null,"WhiteList":null,"Compress":false,"ProxyProtocol":null,"ForwardedHeaders":{"Insecure":true,"TrustedIPs":null}}},"Cluster":null,"Constraints":[],"ACME":{"Email":"dev@example.com","Domains":[{"Main":"foo.example.com","SANs":null}],"Storage":"acme.json","StorageFile":"","OnDemand":false,"OnHostRule":true,"CAServer":"https://acme-v02.api.letsencrypt.org/dcomectory","EntryPoint":"https","KeyType":"","DNSChallenge":null,"HTTPChallenge":{"EntryPoint":"http"},"TLSChallenge":null,"DNSProvider":"","DelayDontCheckDNS":0,"ACMELogging":false,"OverrideCertificates":false,"TLSConfig":null},"DefaultEntryPoints":["https","http"],"ProvidersThrottleDuration":2000000000,"MaxIdleConnsPerHost":200,"IdleTimeout":0,"InsecureSkipVerify":false,"RootCAs":null,"Retry":{"Attempts":0},"HealthCheck":{"Interval":30000000000},"RespondingTimeouts":null,"ForwardingTimeouts":null,"AllowMinWeightZero":false,"Web":null,"Docker":{"Watch":true,"Filename":"","Constraints":null,"Trace":false,"TemplateVersion":2,"DebugLogGeneratedTemplate":false,"Endpoint":"unix:///var/run/docker.sock","Domain":"example.com","TLS":null,"ExposedByDefault":false,"UseBindPortIP":false,"SwarmMode":false,"Network":""},"File":null,"Marathon":null,"Consul":null,"ConsulCatalog":null,"Etcd":null,"Zookeeper":null,"Boltdb":null,"Kubernetes":null,"Mesos":null,"Eureka":null,"ECS":null,"Rancher":null,"DynamoDB":null,"ServiceFabric":null,"Rest":null,"API":{"EntryPoint":"traefik","Dashboard":true,"Debug":true,"CurrentConfigurations":null,"Statistics":null},"Metrics":null,"Ping":null,"HostResolver":null}"
time="2018-10-21T15:30:13Z" level=info msg="nStats collection is disabled.nHelp us improve Traefik by turning this feature on :)nMore details on: https://docs.traefik.io/basics/#collected-datan"
time="2018-10-21T15:30:13Z" level=debug msg="Setting Acme Certificate store from Entrypoint: https"
time="2018-10-21T15:30:13Z" level=info msg="Preparing server http &{Address::80 TLS:<nil> Redcomect:<nil> Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc000505000} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2018-10-21T15:30:13Z" level=info msg="Preparing server https &{Address::443 TLS:0xc0001ea990 Redcomect:<nil> Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc000505020} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2018-10-21T15:30:13Z" level=info msg="Starting server on :80"
time="2018-10-21T15:30:14Z" level=info msg="Preparing server traefik &{Address::8080 TLS:<nil> Redcomect:<nil> Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc000505040} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2018-10-21T15:30:14Z" level=info msg="Starting server on :443"
time="2018-10-21T15:30:14Z" level=info msg="Starting provider configuration.ProviderAggregator {}"
time="2018-10-21T15:30:14Z" level=info msg="Starting server on :8080"
time="2018-10-21T15:30:14Z" level=info msg="Starting provider *docker.Provider {"Watch":true,"Filename":"","Constraints":null,"Trace":false,"TemplateVersion":2,"DebugLogGeneratedTemplate":false,"Endpoint":"unix:///var/run/docker.sock","Domain":"example.com","TLS":null,"ExposedByDefault":false,"UseBindPortIP":false,"SwarmMode":false,"Network":""}"
time="2018-10-21T15:30:14Z" level=info msg="Starting provider *acme.Provider {"Email":"dev@example.com","ACMELogging":false,"CAServer":"https://acme-v02.api.letsencrypt.org/dcomectory","Storage":"acme.json","EntryPoint":"https","KeyType":"","OnHostRule":true,"OnDemand":false,"DNSChallenge":null,"HTTPChallenge":{"EntryPoint":"http"},"TLSChallenge":null,"Domains":[{"Main":"foo.example.com","SANs":null}],"Store":{}}"
time="2018-10-21T15:30:14Z" level=info msg="Testing certificate renew..."
time="2018-10-21T15:30:14Z" level=debug msg="Configuration received from provider ACME: {}"
time="2018-10-21T15:30:14Z" level=debug msg="Provider connection established with docker 17.03.1-ce (API 1.27)"
time="2018-10-21T15:30:14Z" level=debug msg="Looking for provided certificate(s) to validate ["foo.example.com"]..."
time="2018-10-21T15:30:14Z" level=debug msg="No ACME certificate generation requcomed for domains ["foo.example.com"]."
time="2018-10-21T15:30:14Z" level=debug msg="Filtering disabled container /traefik_reverse-proxy_1"
time="2018-10-21T15:30:14Z" level=debug msg="originLabelsmap[traefik.frontend.headers.SSLHost:foo.example.com traefik.frontend.headers.SSLTemporaryRedcomect:true traefik.port:80 com.docker.compose.version:1.18.0 org.label-schema.build-date:2018-10-21T13:17:11Z org.label-schema.vcs-ref:f2246d136931e2b1d757a752b46756845627410f traefik.docker.network:web traefik.frontend.headers.SSLForceHost:true com.docker.compose.container-number:1 com.docker.compose.oneoff:False com.docker.compose.project:web traefik.enable:true com.docker.compose.config-hash:243380275c52ca4cc358b3c4273a1378158361e2dbfe82bf5ef81c6721a4e6ad org.label-schema.vcs-url:https://github.com/example/web.git traefik.frontend.headers.SSLRedcomect:true traefik.frontend.rule:Host:foo.example.com,www.foo.example.com com.docker.compose.service:web maintainer:NGINX Docker Maintainers <docker-maint@nginx.com> traefik.protocol:http]"
time="2018-10-21T15:30:14Z" level=debug msg="allLabelsmap[:map[traefik.frontend.headers.SSLHost:foo.example.com traefik.frontend.headers.SSLTemporaryRedcomect:true traefik.frontend.headers.SSLForceHost:true traefik.frontend.headers.SSLRedcomect:true traefik.frontend.rule:Host:foo.example.com,www.foo.example.com traefik.protocol:http traefik.docker.network:web traefik.port:80 traefik.enable:true]]"

time="2018-10-21T15:30:14Z" level=debug msg="originLabelsmap[traefik.frontend.headers.SSLTemporaryRedcomect:true traefik.port:80 com.docker.compose.version:1.18.0 org.label-schema.build-date:2018-10-21T13:17:11Z org.label-schema.vcs-ref:f2246d136931e2b1d757a752b46756845627410f traefik.docker.network:web traefik.frontend.headers.SSLHost:foo.example.com com.docker.compose.container-number:1 com.docker.compose.oneoff:False com.docker.compose.project:web traefik.enable:true traefik.frontend.headers.SSLForceHost:true com.docker.compose.config-hash:243380275c52ca4cc358b3c4273a1378158361e2dbfe82bf5ef81c6721a4e6ad org.label-schema.vcs-url:https://github.com/example/web.git traefik.frontend.headers.SSLRedcomect:true traefik.frontend.rule:Host:foo.example.com,www.foo.example.com com.docker.compose.service:web maintainer:NGINX Docker Maintainers <docker-maint@nginx.com> traefik.protocol:http]"
time="2018-10-21T15:30:14Z" level=debug msg="allLabelsmap[:map[traefik.frontend.headers.SSLRedcomect:true traefik.enable:true traefik.frontend.rule:Host:foo.example.com,www.foo.example.com traefik.docker.network:web traefik.frontend.headers.SSLHost:foo.example.com traefik.frontend.headers.SSLTemporaryRedcomect:true traefik.port:80 traefik.frontend.headers.SSLForceHost:true traefik.protocol:http]]"
time="2018-10-21T15:30:14Z" level=debug msg="Backend backend-web-web: no load-balancer defined, fallback to 'wrr' method"
time="2018-10-21T15:30:14Z" level=debug msg="Add certificate for domains foo.example.com"
time="2018-10-21T15:30:14Z" level=debug msg="Add certificate for domains foo.example.com,www.foo.example.com"
time="2018-10-21T15:30:14Z" level=info msg="Server configuration reloaded on :80"
time="2018-10-21T15:30:14Z" level=info msg="Server configuration reloaded on :443"
time="2018-10-21T15:30:14Z" level=info msg="Server configuration reloaded on :8080"
time="2018-10-21T15:30:15Z" level=debug msg="Wcoming frontend frontend-Host-foo-example-com-www-foo-example-com-0 to entryPoint https"
time="2018-10-21T15:30:15Z" level=debug msg="Creating backend backend-web-web"
time="2018-10-21T15:30:15Z" level=debug msg="Adding secure middleware for frontend frontend-Host-foo-example-com-www-foo-example-com-0"
time="2018-10-21T15:30:15Z" level=debug msg="Adding TLSClientHeaders middleware for frontend frontend-Host-foo-example-com-www-foo-example-com-0"
time="2018-10-21T15:30:15Z" level=debug msg="Creating load-balancer wrr"
time="2018-10-21T15:30:15Z" level=debug msg="Creating server server-web-web-1-50454c5050fec61e79f56fc0d166926c at http://172.22.0.6:80 with weight 1"
time="2018-10-21T15:30:15Z" level=debug msg="Creating retries max attempts 1"
time="2018-10-21T15:30:15Z" level=debug msg="Creating route route-frontend-Host-foo-example-com-www-foo-example-com-0 Host:foo.example.com,www.foo.example.com"
time="2018-10-21T15:30:15Z" level=debug msg="Wcoming frontend frontend-Host-foo-example-com-www-foo-example-com-0 to entryPoint http"
time="2018-10-21T15:30:15Z" level=debug msg="Creating backend backend-web-web"
time="2018-10-21T15:30:15Z" level=debug msg="Adding secure middleware for frontend frontend-Host-foo-example-com-www-foo-example-com-0"
time="2018-10-21T15:30:15Z" level=debug msg="Adding TLSClientHeaders middleware for frontend frontend-Host-foo-example-com-www-foo-example-com-0"
time="2018-10-21T15:30:15Z" level=debug msg="Creating load-balancer wrr"
time="2018-10-21T15:30:15Z" level=debug msg="Creating server server-web-web-1-50454c5050fec61e79f56fc0d166926c at http://172.22.0.6:80 with weight 1"
time="2018-10-21T15:30:15Z" level=debug msg="Creating retries max attempts 1"
time="2018-10-21T15:30:15Z" level=debug msg="Creating route route-frontend-Host-foo-example-com-www-foo-example-com-0 Host:foo.example.com,www.foo.example.com"
time="2018-10-21T15:30:15Z" level=debug msg="Creating load-balancer wrr"
time="2018-10-21T15:30:15Z" level=debug msg="Creating retries max attempts 1"
time="2018-10-21T15:30:15Z" level=debug msg="Creating load-balancer wrr"
time="2018-10-21T15:30:15Z" level=debug msg="Creating retries max attempts 1"
time="2018-10-21T15:30:15Z" level=debug msg="Add certificate for domains foo.example.com"
time="2018-10-21T15:30:15Z" level=debug msg="Add certificate for domains foo.example.com,www.foo.example.com"
time="2018-10-21T15:30:15Z" level=info msg="Server configuration reloaded on :80"
time="2018-10-21T15:30:15Z" level=info msg="Server configuration reloaded on :443"
time="2018-10-21T15:30:15Z" level=info msg="Server configuration reloaded on :8080"
time="2018-10-21T15:30:15Z" level=debug msg="Try to challenge certificate for domain [foo.example.com www.foo.example.com] founded in Host rule"
time="2018-10-21T15:30:15Z" level=debug msg="Looking for provided certificate(s) to validate ["foo.example.com" "www.foo.example.com"]..."
time="2018-10-21T15:30:15Z" level=debug msg="No ACME certificate generation requcomed for domains ["foo.example.com" "www.foo.example.com"]."

time="2018-10-21T15:30:20Z" level=debug msg="vulcand/oxy/roundrobin/rr: begin ServeHttp on request" Request="{"Method":"HEAD","URL":{"Scheme":"","Opaque":"","User":null,"Host":"","Path":"/","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":""},"Proto":"HTTP/1.1","ProtoMajor":1,"ProtoMinor":1,"Header":{"Accept":["*/*"],"User-Agent":["curl/7.47.0"]},"ContentLength":0,"TransferEncoding":null,"Host":"www.foo.example.com","Form":null,"PostForm":null,"MultipartForm":null,"Trailer":null,"RemoteAddr":"2.176.121.98:48760","RequestURI":"/","TLS":null}"
time="2018-10-21T15:30:20Z" level=debug msg="vulcand/oxy/roundrobin/rr: Forwarding this request to URL" Request="{"Method":"HEAD","URL":{"Scheme":"","Opaque":"","User":null,"Host":"","Path":"/","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":""},"Proto":"HTTP/1.1","ProtoMajor":1,"ProtoMinor":1,"Header":{"Accept":["*/*"],"User-Agent":["curl/7.47.0"]},"ContentLength":0,"TransferEncoding":null,"Host":"www.foo.example.com","Form":null,"PostForm":null,"MultipartForm":null,"Trailer":null,"RemoteAddr":"2.176.121.98:48760","RequestURI":"/","TLS":null}" ForwardURL="http://172.22.0.6:80"
time="2018-10-21T15:30:20Z" level=debug msg="vulcand/oxy/forward: begin ServeHttp on request" Request="{"Method":"HEAD","URL":{"Scheme":"http","Opaque":"","User":null,"Host":"172.22.0.6:80","Path":"","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":""},"Proto":"HTTP/1.1","ProtoMajor":1,"ProtoMinor":1,"Header":{"Accept":["*/*"],"User-Agent":["curl/7.47.0"]},"ContentLength":0,"TransferEncoding":null,"Host":"www.foo.example.com","Form":null,"PostForm":null,"MultipartForm":null,"Trailer":null,"RemoteAddr":"2.176.121.98:48760","RequestURI":"/","TLS":null}"
time="2018-10-21T15:30:20Z" level=debug msg="vulcand/oxy/forward/http: begin ServeHttp on request" Request="{"Method":"HEAD","URL":{"Scheme":"http","Opaque":"","User":null,"Host":"172.22.0.6:80","Path":"","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":""},"Proto":"HTTP/1.1","ProtoMajor":1,"ProtoMinor":1,"Header":{"Accept":["*/*"],"User-Agent":["curl/7.47.0"]},"ContentLength":0,"TransferEncoding":null,"Host":"www.foo.example.com","Form":null,"PostForm":null,"MultipartForm":null,"Trailer":null,"RemoteAddr":"2.176.121.98:48760","RequestURI":"/","TLS":null}"
time="2018-10-21T15:30:20Z" level=debug msg="Upstream ResponseWriter of type *pipelining.writerWithoutCloseNotify does not implement http.CloseNotifier. Returning dummy channel."
time="2018-10-21T15:30:20Z" level=debug msg="vulcand/oxy/forward/http: Round trip: http://172.22.0.6:80, code: 200, Length: 0, duration: 1.908896ms tls:version: 303, tls:resume:false, tls:csuite:c02f, tls:server:www.foo.example.com"
time="2018-10-21T15:30:20Z" level=debug msg="vulcand/oxy/forward/http: completed ServeHttp on request" Request="{"Method":"HEAD","URL":{"Scheme":"http","Opaque":"","User":null,"Host":"172.22.0.6:80","Path":"","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":""},"Proto":"HTTP/1.1","ProtoMajor":1,"ProtoMinor":1,"Header":{"Accept":["*/*"],"User-Agent":["curl/7.47.0"]},"ContentLength":0,"TransferEncoding":null,"Host":"www.foo.example.com","Form":null,"PostForm":null,"MultipartForm":null,"Trailer":null,"RemoteAddr":"2.176.121.98:48760","RequestURI":"/","TLS":null}"
time="2018-10-21T15:30:20Z" level=debug msg="vulcand/oxy/forward: completed ServeHttp on request" Request="{"Method":"HEAD","URL":{"Scheme":"http","Opaque":"","User":null,"Host":"172.22.0.6:80","Path":"","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":""},"Proto":"HTTP/1.1","ProtoMajor":1,"ProtoMinor":1,"Header":{"Accept":["*/*"],"User-Agent":["curl/7.47.0"]},"ContentLength":0,"TransferEncoding":null,"Host":"www.foo.example.com","Form":null,"PostForm":null,"MultipartForm":null,"Trailer":null,"RemoteAddr":"2.176.121.98:48760","RequestURI":"/","TLS":null}"
time="2018-10-21T15:30:20Z" level=debug msg="vulcand/oxy/roundrobin/rr: completed ServeHttp on request" Request="{"Method":"HEAD","URL":{"Scheme":"","Opaque":"","User":null,"Host":"","Path":"/","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":""},"Proto":"HTTP/1.1","ProtoMajor":1,"ProtoMinor":1,"Header":{"Accept":["*/*"],"User-Agent":["curl/7.47.0"]},"ContentLength":0,"TransferEncoding":null,"Host":"www.foo.example.com","Form":null,"PostForm":null,"MultipartForm":null,"Trailer":null,"RemoteAddr":"2.176.121.98:48760","RequestURI":"/","TLS":null}"


```

**Comment by m-nouredini** [comment]
**Update:**
Today I removed `[acme]` configuration and now everything seems to work correctly.
I'm not very familiar with Traefik code base, but I spent a day reading pull requests and issues related to this feature and took a look at `/unrolled/secure/secure.go`. according to [unrolled/pull/35](https://github.com/unrolled/secure/pull/35) the module is tested and works fine, so I guess that the `unrolled` module does not handle SSL requests if `acme` is enabled.

**Comment by m-nouredini** [comment]
**Update:**
Today I removed `[acme]` configuration and now everything seems to work correctly.
I'm not very familiar with Traefik code base, but I spent a day reading pull requests and issues related to this feature and took a look at `/unrolled/secure/secure.go`. according to [unrolled/pull/35](https://github.com/unrolled/secure/pull/35) the module is tested and works fine, so I guess that the `unrolled` module does not handle SSL requests if `acme` is enabled.

**Comment by m-nouredini** [comment]
**Update:**
Today I removed `[acme]` configuration and now everything seems to work correctly.
I'm not very familiar with Traefik code base, but I spent a day reading pull requests and issues related to this feature and took a look at `/unrolled/secure/secure.go`. according to [unrolled/pull/35](https://github.com/unrolled/secure/pull/35) the module is tested and works fine, so I guess that the `unrolled` module does not handle SSL requests if `acme` is enabled.

**Comment by m-nouredini** [comment]
### **Update:**
Today I removed `[acme]` configuration and now everything seems to work correctly.
I'm not very familiar with Traefik code base, but I spent a day reading pull requests and issues related to this feature and took a look at `/unrolled/secure/secure.go`. according to [unrolled/pull/35](https://github.com/unrolled/secure/pull/35) the module is tested and works fine, so I guess that the `unrolled` module does not handle SSL requests if `acme` is enabled.

**Comment by m-nouredini** [comment]
I think the problem is because SSLForceHost is not set for the front end, as you can see in the following image:
![image](https://user-images.githubusercontent.com/3660120/47353660-a7fdd180-d6c9-11e8-98f8-1a8e36d59e29.png)

I've set `traefik.frontend.headers.SSLForceHost=true` label for the container but it seems that it is set as `SSL Host` value

**Comment by ldez** [comment]
It's just a typo in the UI :wink:

**Comment by Ullaakut** [comment]
Hi @m-nouredini !


Thanks for reporting this issue. I tried replicating your problem, and with the exact same configuration and labels, I seem to get the expected behavior and am unable to reproduce your issue.

Could you show me what `curl` command you ran to get that 200 OK response without a redirection?

I personally get this output when running this `curl` command.

`$> curl -v -H Host:www.foo.example.com 127.0.0.1/`

```
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 80 (#0)
> GET / HTTP/1.1
> Host:www.foo.example.com
> User-Agent: curl/7.54.0
> Accept: */*
>
< HTTP/1.1 301 Moved Permanently
< Content-Type: text/html; charset=utf-8
< Location: https://foo.example.com/
< Date: Wed, 24 Oct 2018 13:03:12 GMT
< Content-Length: 59
<
<a href="https://foo.example.com/">Moved Permanently</a>.

* Connection #0 to host 127.0.0.1 left intact
```

As you can see the request is redirected to `https://foo.example.com/` like your configuration specifies, and I get a `301 Moved Permanently` response status.

**Comment by m-nouredini** [comment]
Hello @Ullaakut !
Thanks for your attention, and sorry for my late response, 
You're right but my problem is with SSL protocol, I get 301 for http too, but it's not working for https as you can see below:
http:
`curl -I http://www.foo.example.com`
```
HTTP/1.1 301 Moved Permanently
Content-Type: text/html; charset=utf-8
Location: https://foo.example.com/
Date: Sat, 27 Oct 2018 07:30:17 GMT
```
https:
`$> curl -I https://www.foo.example.com`
```
HTTP/1.1 200 OK
Accept-Ranges: bytes
Content-Length: 612
Content-Type: text/html
Date: Sat, 27 Oct 2018 07:21:12 GMT
Etag: "5a1437f4-264"
Last-Modified: Tue, 21 Nov 2017 14:28:04 GMT
Server: nginx/1.13.7
```
and this is the traefik log for https request:
```
time="2018-10-27T07:21:12Z" level=debug msg="vulcand/oxy/roundrobin/rr: begin ServeHttp on request" Request="{"Method":"HEAD","URL":{"Scheme":"","Opaque":"","User":null,"Host":"","Path":"/","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":""},"Proto":"HTTP/1.1","ProtoMajor":1,"ProtoMinor":1,"Header":{"Accept":["*/*"],"User-Agent":["curl/7.47.0"]},"ContentLength":0,"TransferEncoding":null,"Host":"www.foo.example.com","Form":null,"PostForm":null,"MultipartForm":null,"Trailer":null,"RemoteAddr":"172.20.0.1:52100","RequestURI":"/","TLS":null}"
time="2018-10-27T07:21:12Z" level=debug msg="vulcand/oxy/roundrobin/rr: Forwarding this request to URL" ForwardURL="http://172.20.0.3:80" Request="{"Method":"HEAD","URL":{"Scheme":"","Opaque":"","User":null,"Host":"","Path":"/","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":""},"Proto":"HTTP/1.1","ProtoMajor":1,"ProtoMinor":1,"Header":{"Accept":["*/*"],"User-Agent":["curl/7.47.0"]},"ContentLength":0,"TransferEncoding":null,"Host":"www.foo.example.com","Form":null,"PostForm":null,"MultipartForm":null,"Trailer":null,"RemoteAddr":"172.20.0.1:52100","RequestURI":"/","TLS":null}"
time="2018-10-27T07:21:12Z" level=debug msg="vulcand/oxy/forward: begin ServeHttp on request" Request="{"Method":"HEAD","URL":{"Scheme":"http","Opaque":"","User":null,"Host":"172.20.0.3:80","Path":"","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":""},"Proto":"HTTP/1.1","ProtoMajor":1,"ProtoMinor":1,"Header":{"Accept":["*/*"],"User-Agent":["curl/7.47.0"]},"ContentLength":0,"TransferEncoding":null,"Host":"www.foo.example.com","Form":null,"PostForm":null,"MultipartForm":null,"Trailer":null,"RemoteAddr":"172.20.0.1:52100","RequestURI":"/","TLS":null}"
time="2018-10-27T07:21:12Z" level=debug msg="vulcand/oxy/forward/http: begin ServeHttp on request" Request="{"Method":"HEAD","URL":{"Scheme":"http","Opaque":"","User":null,"Host":"172.20.0.3:80","Path":"","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":""},"Proto":"HTTP/1.1","ProtoMajor":1,"ProtoMinor":1,"Header":{"Accept":["*/*"],"User-Agent":["curl/7.47.0"]},"ContentLength":0,"TransferEncoding":null,"Host":"www.foo.example.com","Form":null,"PostForm":null,"MultipartForm":null,"Trailer":null,"RemoteAddr":"172.20.0.1:52100","RequestURI":"/","TLS":null}"
time="2018-10-27T07:21:12Z" level=debug msg="Upstream ResponseWriter of type *pipelining.writerWithoutCloseNotify does not implement http.CloseNotifier. Returning dummy channel."
time="2018-10-27T07:21:12Z" level=debug msg="vulcand/oxy/forward/http: Round trip: http://172.20.0.3:80, code: 200, Length: 0, duration: 2.463685ms tls:version: 303, tls:resume:false, tls:csuite:c02f, tls:server:www.foo.example.com"
time="2018-10-27T07:21:12Z" level=debug msg="vulcand/oxy/forward/http: completed ServeHttp on request" Request="{"Method":"HEAD","URL":{"Scheme":"http","Opaque":"","User":null,"Host":"172.20.0.3:80","Path":"","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":""},"Proto":"HTTP/1.1","ProtoMajor":1,"ProtoMinor":1,"Header":{"Accept":["*/*"],"User-Agent":["curl/7.47.0"]},"ContentLength":0,"TransferEncoding":null,"Host":"www.foo.example.com","Form":null,"PostForm":null,"MultipartForm":null,"Trailer":null,"RemoteAddr":"172.20.0.1:52100","RequestURI":"/","TLS":null}"
time="2018-10-27T07:21:12Z" level=debug msg="vulcand/oxy/forward: completed ServeHttp on request" Request="{"Method":"HEAD","URL":{"Scheme":"http","Opaque":"","User":null,"Host":"172.20.0.3:80","Path":"","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":""},"Proto":"HTTP/1.1","ProtoMajor":1,"ProtoMinor":1,"Header":{"Accept":["*/*"],"User-Agent":["curl/7.47.0"]},"ContentLength":0,"TransferEncoding":null,"Host":"www.foo.example.com","Form":null,"PostForm":null,"MultipartForm":null,"Trailer":null,"RemoteAddr":"172.20.0.1:52100","RequestURI":"/","TLS":null}"
time="2018-10-27T07:21:12Z" level=debug msg="vulcand/oxy/roundrobin/rr: completed ServeHttp on request" Request="{"Method":"HEAD","URL":{"Scheme":"","Opaque":"","User":null,"Host":"","Path":"/","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":""},"Proto":"HTTP/1.1","ProtoMajor":1,"ProtoMinor":1,"Header":{"Accept":["*/*"],"User-Agent":["curl/7.47.0"]},"ContentLength":0,"TransferEncoding":null,"Host":"www.foo.example.com","Form":null,"PostForm":null,"MultipartForm":null,"Trailer":null,"RemoteAddr":"172.20.0.1:52100","RequestURI":"/","TLS":null}"

```
By the way, my problem is fixed for now, because I decided to use my old certificates and removed the `acme` part of `traefik.toml` and with my current configuration everything seems to work fine, I think there is a problem with let's encrypt plugin or some other modules which handles ssl requests when `acme` is enabled.
this is my current config which works :
```toml
debug = true

defaultEntryPoints = ["https","http"]

[api]
  dashboard=true

[entryPoints]
  [entryPoints.http]
  address = ":80"
  [entryPoints.https]
  address = ":443"
  [entryPoints.https.tls]
    [[entryPoints.https.tls.certificates]]
    certFile = "/certs/fullchain.pem"
    keyFile = "/certs/privkey.pem"

[retry]

[docker]
endpoint = "unix:///var/run/docker.sock"
domain = "example.com"
watch = true
exposedByDefault = false

```
It'd be great if we can have both `SSLForceHost` and let's encrypt auto-renew feature, So if you need more info, I'll be glad to help!

**Comment by Ullaakut** [comment]
Thanks for the precisions, and my bad, I should have thought about trying on the `https` endpoint rather than the `http`. I'll try again this morning and hopefully will also find the source of the issue!

**Comment by traefiker** [comment]
Closed by #4138.

**Comment by aek** [comment]
Thank you @mmatur !! I have waiting to be able to use this so long. Good catch
